plugins/module_utils/models.py

```diff
@@ -82,17 +82,32 @@ def from_json(cls, json: TerraformModuleResource) -> "TerraformAnsibleProvider":
 @dataclass
 class TerraformChildModule:
     resources: List[TerraformModuleResource]
+    child_modules: List["TerraformChildModule"]
```

Comment:
```suggestion
    child_modules: List["TerraformChildModule"] = field(default_factory=lambda: [])
```
You need to define default value  for this parameter in order for unit tests to pass (the following import `from dataclasses import dataclass` should be updated to `from dataclasses import dataclass, field`).